extension/background.js

```diff
@@ -0,0 +1,195 @@
+"use strict";
+
+const TELEMETRY_CATEGORY = "addonsSearchExperiment";
+// methods
+const TELEMETRY_METHOD_ETLD_CHANGE = "etld_change";
+// objects
+const TELEMETRY_OBJECT_WEBREQUEST = "webrequest";
+const TELEMETRY_OBJECT_OTHER = "other";
+// values
+const TELEMETRY_VALUE_EXTENSION = "extension";
+const TELEMETRY_VALUE_SERVER = "server";
+
+class AddonsSearchExperiment {
+  constructor() {
+    this.matchPatternsMap = {};
+
+    console.debug("registering telemetry events");
+    browser.telemetry.registerEvents(TELEMETRY_CATEGORY, {
+      [TELEMETRY_METHOD_ETLD_CHANGE]: {
+        methods: [TELEMETRY_METHOD_ETLD_CHANGE],
+        objects: [TELEMETRY_OBJECT_WEBREQUEST, TELEMETRY_OBJECT_OTHER],
+        extra_keys: ["addonId", "addonVersion", "from", "to"],
+        record_on_release: true,
+      },
+    });
+  }
+
+  async getMatchPatterns() {
+    try {
+      this.matchPatternsMap =
+        await browser.addonsSearchExperiment.getMatchPatterns();
+    } catch (err) {
+      console.error(`failed to retrieve the list of URL patterns: ${err}`);
+      this.matchPatternsMap = {};
+    }
+
+    return this.matchPatternsMap;
+  }
+
+  async monitor() {
```

Comment:
Add a comment explaining this function.  Somethign along the lines "If the search service changes the set of engines that are enabled, we update our pattern matching in the webrequest listeners"